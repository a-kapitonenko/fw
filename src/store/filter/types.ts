import { Frame } from '../frames/types';

export type Field = {
  name: string;
  value: string;
  checked?: boolean;
};

export type Groups = {
  color: Field[];
  width: Field[];
  noseBridge: Field[];
  shape: Field[];
  material: Field[];
};

export const enum FilterActionTypes {
  FETCH_REQUEST = '@@filter/FETCH_REQUEST',
  FETCH_SUCCESS = '@@filter/FETCH_SUCCESS',
  FETCH_ERROR = '@@filter/FETCH_ERROR',
  CHANGE_CHECKED = '@@filter/CHANGE_CHECKED',
};

export interface IFilterState {
  readonly fetching: boolean;
  readonly groups: Groups;
  readonly errors: string;
  readonly query: Object;
  readonly frames: Frame[];
};
