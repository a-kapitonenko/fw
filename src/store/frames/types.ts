export type Frame = {
  upc: number;
  label: string;
  value: string; 
  img: string;
  compatibility?: boolean;
};

export const enum FramesActionTypes {  
  SUBMIT_START = '@@frames/SUBMIT_START',
  SUBMIT_SUCCESS = '@@frames/SUBMIT_SUCCESS',
  SUBMIT_FAILED = '@@frames/SUBMIT_FAILED',
  FETCH_SIMILAR_FRAMES_START = '@@frames/FETCH_SIMILAR_FRAMES_START',
  FETCH_SIMILAR_FRAMES_SUCCESS = '@@frames/FETCH_SIMILAR_FRAMES_SUCCESS',
  FETCH_SIMILAR_FRAMES_FAILED = '@@frames/FETCH_SIMILAR_FRAMES_FAILED',
  CHECK_FRAME_START = '@@frames/CHECK_FRAME_START',
  CHECK_FRAME_SUCCESS = '@@frames/CHECK_FRAME_SUCCESS',
  CHECK_FRAME_FAILED = '@@frames/CHECK_FRAME_FAILED',
  OPEN = '@@frames/OPEN',
  CLOSE = '@@frames/CLOSE',
  SET_SELECTED_FRAME = '@@frames/SET_SELECTED_FRAME',
  CLEAR_SELECTED_FRAME = '@@frame/CLEAR_SELECTED_FRAME',
  CLEAR_RESULT = '@@frames/CLEAR_RESULT',
};

export interface FramesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly open: boolean;
  readonly selectedFrame: Frame;
  readonly similarFrames: {
    readonly isFetching: boolean;
    readonly errors: string;
    readonly data: Frame[];
  };
};
