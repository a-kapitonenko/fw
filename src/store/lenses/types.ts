export type Lens = {
  name: string;
  value: number;
};

export const enum LensesActionTypes {
  FETCH_LENSES_START = '@@lenses/FETCH_LENSES_START',
  FETCH_LENSES_SUCCESS = '@@lenses/FETCH_LENSES_SUCCESS',
  FETCH_LENSES_FAILED = '@@lenses/FETCH_LENSES_FAILED',

  SAVE_LENS_START = '@@lenses/SAVE_LENS_START',
  SAVE_LENS_SUCCESS = '@@lenses/SAVE_LENS_SUCCESS',
  SAVE_LENS_FAILED = '@@lenses/SAVE_LENS_FAILED',

  CHECK_LENS_START = '@@lenses/CHECK_LENS_START',
  CHECK_LENS_SUCCESS = '@@lenses/CHECK_LENS_SUCCESS',
  CHECK_LENS_FALIED = '@@lenses/CHECK_LENS_FAILED',
};

export interface ILensesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly lenses: Lens[];
};
