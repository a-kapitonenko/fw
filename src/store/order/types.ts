import { Frame } from '../frames/types';
import { Lens } from '../lenses/types';

export type OculusInfo = {
  sphere: string;
  cyclinder: string;
  axis: string;
  addition: string;
  PDDistance: string;
  PDNear: string;
  prism: string;
};

export type Prescription = {
  OD: OculusInfo;
  OS: OculusInfo;
};

export type Blueprint = {
  img: string;
};

export type Barcode = {
  img: string;
}

export const enum OrderActionTypes {
  SET_RX_INFORMATION = '@@order/SET_RX_INFORMATION',
  SET_RECOMMENDATION = '@@order/SET_RECOMMENDATION',
  SET_LENS = '@@order/SET_LENS',
  SET_FRAME = '@@order/SET_FRAME',
  SET_MESSAGE = '@@order/SET_MESSAGE',
  SET_FITTING_PROPERTIES = '@@order/SET_FITTING_PROPERTIES',
  SET_FITTING_HEIGHT = '@@order/SET_FITTING_HEIGHT',
  SET_BLUEPRINT = '@@order/SET_BLUEPRINT',
  SET_BARCODE = '@@order/SET_BARCODE',
  SET_ERRORS = '@@order/SET_ERRORS',
  DELETE_ERRORS = '@@order/DELETE_ERRORS'
};

export interface IOrderState {
  readonly prescription: Prescription;
  readonly frame: Frame;
  readonly fittingHeight: number;
  readonly fittingProperties: any;
  readonly lens: Lens;
  readonly recommendation: string;
  readonly message: string;
  readonly blueprint: Blueprint;
  readonly barcode: Barcode;
  readonly errors: object;
};
