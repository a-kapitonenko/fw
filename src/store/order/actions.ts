import { action } from 'typesafe-actions';
import { ApplicationState } from '../index';
import { OrderActionTypes, Prescription, Blueprint, BossTypes, Boss } from './types';

export const submitStart = (boss: Boss) => action(OrderActionTypes.SUBMIT_START, boss);
export const submitSuccess = () => action(OrderActionTypes.SUBMIT_SUCCESS);
export const submitFailed = (message: string) => action(OrderActionTypes.SUBMIT_FAILED, message);

export const saveOrderStart = (state: ApplicationState) => action(OrderActionTypes.SAVE_ORDER_START, state);
export const saveOrderSuccess = () => action(OrderActionTypes.SAVE_ORDER_SUCCESS);
export const saveOrderFailed = (message: string) => action(OrderActionTypes.SAVE_ORDER_FAILED, message);

export const fetchOrderValuesStart = (id: string) => action(OrderActionTypes.FETCH_ORDER_VALUES_START, id);
export const fetchOrderValuesSuccess = (values: any) => action(OrderActionTypes.FETCH_ORDER_VALUES_SUCCESS, values);
export const fetchOrderValuesFailed = (message: string) => action(OrderActionTypes.FETCH_ORDER_VALUES_FAILED, message);

export const savePrescriptionStart = (prescription: Prescription) => action(OrderActionTypes.SAVE_PRESCRIPTION_START, prescription);
export const savePrescriptionSuccess = () => action(OrderActionTypes.SAVE_PRESCRIPTION_SUCCESS);
export const savePrescriptionFailed = (message: string) => action(OrderActionTypes.SAVE_PRESCRIPTION_FAILED, message);

export const saveFittingHeightStart = (boss: Boss, height: number) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_START, { boss, height });
export const saveFittingHeightSuccess = (blueprint: Blueprint) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_SUCCESS, blueprint);
export const saveFittingHeightFailed = (message: string) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_FAILED, message);

export const setRxInformation = (type: string, field: string, value: string) => action(OrderActionTypes.SET_RX_INFORMATION, { type, field, value });
export const setRecommendation = (recommendation: string) => action(OrderActionTypes.SET_RECOMMENDATION, recommendation);
export const setMessage = (message: string) => action(OrderActionTypes.SET_MESSAGE, message);
export const setFittingProperties = (properties: any) => action(OrderActionTypes.SET_FITTING_PROPERTIES, properties);
export const setBoss = (type: BossTypes, value: any) => action(OrderActionTypes.SET_BOSS, { type, value });
export const disableRedirect = () => action(OrderActionTypes.DISABLE_REDIRECT);
