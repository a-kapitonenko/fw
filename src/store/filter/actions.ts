import { action } from 'typesafe-actions';
import { FilterActionTypes, Groups } from './types';
import { Boss } from '../order/types';
import { Frame } from '../frames/types';

export const filteringStart = (boss: Boss, query: object) => action(FilterActionTypes.FILTERING_START, { boss, query });
export const filteringSuccess = (frames: Frame[]) => action(FilterActionTypes.FILTERING_SUCCESS, frames);
export const filteringFailed = (message: string) => action(FilterActionTypes.FILTERING_FAILED, message);

export const fetchGroupsStart = () => action(FilterActionTypes.FETCH_GROUPS_START);
export const fetchGroupsSuccess = (groups: Groups) => action(FilterActionTypes.FETCH_GROUPS_SUCCESS, groups);
export const fetchGroupsFailed = (message: string) => action(FilterActionTypes.FETCH_GROUPS_FAILED, message);

export const changeChecked = (type: string, name: string, value: boolean) => action(FilterActionTypes.CHANGE_CHECKED, { type, name, value });
export const clearChecked = () => action(FilterActionTypes.CLEAR_CHECKED);
export const addQuery = (type: string, value: any) => action(FilterActionTypes.ADD_QUERY, { type, value });
export const deleteQuery = (type: string, value: any) => action(FilterActionTypes.DELETE_QUERY, { type, value });
export const clearQuery = () => action(FilterActionTypes.CLEAR_QUERY);
export const clearResult = () => action(FilterActionTypes.CLEAR_RESULT);
