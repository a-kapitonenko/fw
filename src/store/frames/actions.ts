import { Dispatch } from 'redux';
import { action } from 'typesafe-actions';

import { FramesActionTypes, Frame } from './types';
import { IOrderState } from '../order/types';
import * as OrderActions from '../order/actions';

import { getSimilarFrames, onSubmit } from '../../test/frames';

export const fetchRequest = () => action(FramesActionTypes.FETCH_REQUEST);
export const fetchSuccess = (similarFrames: Frame[]) => action(FramesActionTypes.FETCH_SUCCESS, similarFrames);
export const fetchError = (message: string) => action(FramesActionTypes.FETCH_ERROR, message);
export const open = () => action(FramesActionTypes.OPEN);
export const close = () => action(FramesActionTypes.CLOSE);
export const setStep = (step: number) => action(FramesActionTypes.SET_STEP, step);
export const setSelectedFrame = (frame: Frame) => action(FramesActionTypes.SET_SELECTED_FRAME, frame);

export const fetchSimilarFrames: any = (order: IOrderState) => (dispatch: Dispatch) => {
  dispatch(fetchRequest());

  return new Promise((resolver) => {
    const response = getSimilarFrames(order);

    resolver(response);
  })
  .then((response: any) => {
    if (response.success) {
      dispatch(fetchSuccess(response.similarFrames));
    }
  })
};

export const fetchSubmit: any = (order: IOrderState, frame: Frame) => (dispatch: Dispatch) => {
  return new Promise((resolver) => {
    const response = onSubmit(order, frame);

    resolver(response);
  })
  .then((response: any) => {
    if (response.success) {
      dispatch(close());
      dispatch(OrderActions.setFrame(frame));
      dispatch(setStep(1));
      dispatch(OrderActions.setMessage(response.result.message));
      dispatch(OrderActions.setFittingProperties(response.result.fittingHeight));
    }
  })
};
