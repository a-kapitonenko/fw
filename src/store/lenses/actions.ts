import { Dispatch } from 'redux';
import { action } from 'typesafe-actions';

import { LensesActionTypes, Lens } from './types';
import { IOrderState, Prescription, BossTypes } from '../order/types';
import * as orderActions from '../order/actions';

import { getLenses } from '../../test/lenses';
import { saveLensInformation } from '../../test/order';

// export const fetchRequest = () => action(LensesActionTypes.FETCH_REQUEST);
// export const closeReauest = () => action(LensesActionTypes.CLOSE_REQUEST);
export const setErrors = (errors: string) => action(LensesActionTypes.SET_ERRORS, errors);
export const clearErrors = () => action(LensesActionTypes.CLEAR_ERRORS);
export const setLenses = (lenses: Lens[]) => action(LensesActionTypes.SET_LENSES, lenses);

export const fetchLenses: any = (order: IOrderState) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = getLenses(order);
    
    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(setLenses(response.lenses));
    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};

export const saveLens: any = (prescription: Prescription, lens: Lens) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = saveLensInformation(prescription, lens)

    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(orderActions.setRecommendation(response.recommendation));
      dispatch(orderActions.setBoss(BossTypes.LENS, lens));

    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};
