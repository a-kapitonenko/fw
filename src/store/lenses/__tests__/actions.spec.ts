import * as lensesActions from '../actions';
import { Prescription } from '../../order/types';
import { Lens } from '../types';
import * as mockData from '../../../mockData';

describe('lenses actions', () => {
  it('test fetchLensesStart action creator', () => {
    const prescription: Prescription = mockData.prescription;

    expect(lensesActions.fetchLensesStart(prescription)).toMatchSnapshot();
  });

  it('test fetchLensesSuccess action creator', () => {
    const lenses: Lens[] = mockData.lenses;

    expect(lensesActions.fetchLensesSuccess(lenses)).toMatchSnapshot();
  });

  it('test fetchLensesFailed action creator', () => {
    const message: string = mockData.errorMessage;

    expect(lensesActions.fetchLensesFailed(message)).toMatchSnapshot();
  });

  it('test saveLensStart action creator', () => {
    const prescription: Prescription = mockData.prescription;
    const lens: Lens = mockData.lens;

    expect(lensesActions.saveLensStart(prescription, lens)).toMatchSnapshot();
  });

  it('test saveLensSuccess action creator', () => {
    expect(lensesActions.saveLensSuccess()).toMatchSnapshot();
  });

  it('test saveLensFailed action creator', () => {
    const message: string = mockData.errorMessage;

    expect(lensesActions.saveLensFailed(message)).toMatchSnapshot();
  });

  it('test checkLensStart action creator', () => {
    const prescription: Prescription = mockData.prescription;
    const lens: Lens = mockData.lens;

    expect(lensesActions.checkLensStart(prescription, lens)).toMatchSnapshot();
  });

  it('test checkLensSuccess action creator', () => {
    expect(lensesActions.checkLensSuccess()).toMatchSnapshot();
  });

  it('test checkLensFailed action creator', () => {
    const message: string = mockData.errorMessage;

    expect(lensesActions.checkLensFailed(message)).toMatchSnapshot();
  });

  it('test setInitialValue action creator', () => {
    const lenses: Lens[] = mockData.lenses;

    expect(lensesActions.setInitialValue(lenses)).toMatchSnapshot();
  });
});
