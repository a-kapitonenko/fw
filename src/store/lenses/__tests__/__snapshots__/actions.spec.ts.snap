// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lenses actions test checkLensFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@lenses/CHECK_LENS_FAILED",
}
`;

exports[`lenses actions test checkLensStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "lens": Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown",
      "value": 1,
    },
    "prescription": Object {
      "OD": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "2",
        "axis": "1",
        "cyclinder": "2",
        "prism": "1",
        "sphere": "1",
      },
      "OS": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "2",
        "axis": "1",
        "cyclinder": "2",
        "prism": "1",
        "sphere": "1",
      },
    },
  },
  "type": "@@lenses/CHECK_LENS_START",
}
`;

exports[`lenses actions test checkLensSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@lenses/CHECK_LENS_SUCCESS",
}
`;

exports[`lenses actions test fetchLensesFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@lenses/FETCH_LENSES_FAILED",
}
`;

exports[`lenses actions test fetchLensesStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "OD": Object {
      "PDDistance": "1",
      "PDNear": "1",
      "addition": "2",
      "axis": "1",
      "cyclinder": "2",
      "prism": "1",
      "sphere": "1",
    },
    "OS": Object {
      "PDDistance": "1",
      "PDNear": "1",
      "addition": "2",
      "axis": "1",
      "cyclinder": "2",
      "prism": "1",
      "sphere": "1",
    },
  },
  "type": "@@lenses/FETCH_LENSES_START",
}
`;

exports[`lenses actions test fetchLensesSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Array [
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown",
      "value": 1,
    },
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Transitions* Signature VII",
      "value": 2,
    },
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- XTRActive* Gray",
      "value": 3,
    },
  ],
  "type": "@@lenses/FETCH_LENSES_SUCCESS",
}
`;

exports[`lenses actions test saveLensFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@lenses/SAVE_LENS_FAILED",
}
`;

exports[`lenses actions test saveLensStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "lens": Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown",
      "value": 1,
    },
    "prescription": Object {
      "OD": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "2",
        "axis": "1",
        "cyclinder": "2",
        "prism": "1",
        "sphere": "1",
      },
      "OS": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "2",
        "axis": "1",
        "cyclinder": "2",
        "prism": "1",
        "sphere": "1",
      },
    },
  },
  "type": "@@lenses/SAVE_LENS_START",
}
`;

exports[`lenses actions test saveLensSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@lenses/SAVE_LENS_SUCCESS",
}
`;

exports[`lenses actions test setInitialValue action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Array [
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown",
      "value": 1,
    },
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Transitions* Signature VII",
      "value": 2,
    },
    Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- XTRActive* Gray",
      "value": 3,
    },
  ],
  "type": "@@lenses/SET_INITIAL_VALUE",
}
`;
